### DISCUSSION TITLE
What is "Fixed Registration Address for Agent Nodes" for?

### DISCUSSION THREAD

**Discussion by lesomnus** [k3s-io/k3s#10991] [Q&A] [answered]
What is the reason for using a [Fixed Registration Address](https://docs.k3s.io/architecture#fixed-registration-address-for-agent-nodes)?
Isn't it possible for the agent nodes to send requests to the alive api-server nodes through load balancing, even if any api-server node goes down?

Or is this setting:
1. used only when joining the agent nodes to the cluster for the first time?  Once the agent nodes are joined to the cluster, do they forget this value?
1. intended to connect the agent nodes only to specific server nodes?
1. considering the scenario where all server nodes are replaced at once?

**Comment by brandond** [comment, marked as answer]
1. No, they do not forget it, it is used as a fallback if direct connections to servers fail for whatever reason.
2. No, agents always discover the full server list from the cluster and will fail over between them.
3. Yes, that's the idea. If you replace all of the servers that the agent is aware of, it will fall back to the fixed registration endpoint. If you just point the agent directly at a server, you'd need to go update the config on all the agents to point at a new server.
